maven-core/src/main/java/org/apache/maven/toolchain/DefaultToolchainManager.java

```diff
@@ -43,12 +46,25 @@
 public class DefaultToolchainManager
     implements ToolchainManager
 {
-    @Inject
-    Logger logger;
+    protected final Logger logger;
 
     @Inject
     Map<String, ToolchainFactory> factories;
 
+    @Inject
+    public DefaultToolchainManager()
+    {
+        this( LoggerFactory.getLogger( DefaultToolchainManager.class ) );
+    }
+
+    /**
+     * Ctor for UT.
+     */
+    DefaultToolchainManager( Logger logger )
+    {
+        this.logger = requireNonNull( logger );
```

Comment:
I wonder why UTs need this....

Reply:
I did not touch UTs, I merely "cleaned up core logging". Still, added a TODO, as in default toolchain manager (and private) components there are some peculiarities: annotated component should not be extended (is bad practice), so IMO this needs some refactoring...